### PR TITLE
Feat: duplicate table

### PR DESCRIPTION
**Pull request by lanzoninicola**
The new feature adds an item to the contextual menu that appears when the user right-clicks on a table. This item allows the user to duplicate the table, copying both the structure and the data (except indexes, triggers, constraints) (it works doing a "select * into <new_table> from <table_name>).

The video that has been provided demonstrates how the feature works in action, allowing users to see exactly how the new functionality will benefit them. Overall, this feature should make it much easier and more efficient for users to duplicate tables, saving them time and effort in their database editing work.

Link to the video: https://youtu.be/cn0dG4YSIZk

Furthermore, it is worth mentioning that in the future, there is also a possibility to add a flag that allows the user to choose how much data to copy when duplicating a table. Specifically, the flag could enable the user to copy all data, only a certain number of records (X records), or no data at all. This added functionality would provide even more flexibility and control to users when using the duplicate table feature.

Tested with MYSQL, POSTGRES, MSSQL.

Thanks

### CODE REVIEW

File: apps/studio/src/components/CoreTabs.vue [hunk 1]
```diff
@@ -238,6 +263,48 @@
...
+
+        this.$modal.hide(this.duplicateTableModal)
+
+        this.$nextTick(async() => {
+          try {
+            if (this.dbAction.toLowerCase() !== 'duplicate') {
+              return
+            }
+
+            await this.connection.duplicateTable(tableName, this.duplicatedTableName, schema)
```
Comment (rathboma) [inline apps/studio/src/components/CoreTabs.vue:292]: Do you need to clear duplicatesTableName and duplicatedTableParams here in a finally block?

File: apps/studio/src/lib/db/client.ts [hunk 2]
```diff
@@ -82,6 +82,9 @@ export interface DatabaseClient {
   // delete stuff
   dropElement: (elementName: string, typeOfElement: DatabaseElement, schema?: string) => Promise<void>
   truncateElement: (elementName: string, typeOfElement: DatabaseElement, schema?: string) => Promise<void>
+
+  // duplicate table
```
Comment (rathboma) [inline apps/studio/src/lib/db/client.ts:86]: Could you possibly also add a ‘duplicateTableSql’ method here (similar to alterTableSql vs alterTable) so users can edit the sql themselves if they like?

File: apps/studio/src/lib/db/clients/mysql.js [hunk 3]
```diff
@@ -755,6 +759,14 @@ export async function truncateElement (conn, elementName, typeOfElement) {
   });
 }
 
+export async function duplicateTable(conn, tableName, newTable) {
+  let sql = `CREATE TABLE ${wrapIdentifier(newTable)} LIKE ${wrapIdentifier(tableName)};`;
```
Comment (rathboma) [inline apps/studio/src/lib/db/clients/mysql.js:763]: I didn’t even know about this syntax! Thanks for this

**Review by rathboma — CHANGES_REQUESTED** [review]
Fantastic PR thanks so much!

I’ve added a couple of requests inline with the code.

one last ask:
would you be able to add an integration test in the db.ts and all.is file? That way we can make sure it works for all databases we add going forward :-)

File: apps/studio/src/components/CoreTabs.vue [hunk 4]
```diff
@@ -73,6 +73,24 @@
...
+          <div class="form-group">
+            <label for="duplicatedTableName">New table name</label>
+            <input type="text" name="duplicatedTableName" class="form-control" required v-model="duplicatedTableName" autofocus>
+          </div>
+          <small>* Indexes, relations or triggers will not be duplicated</small>
+        </div>
+        <div class="vue-dialog-buttons">
+          <span class="expand"></span>
+          <button ref="no" @click.prevent="$modal.hide(duplicateTableModal)" class="btn btn-sm btn-flat">Cancel</button>
+          <button @focusout="sureOpen && $refs.no && $refs.no.focus()" @click.prevent="completeDuplicateTableAction" class="btn btn-sm btn-primary">{{this.titleCaseAction}} {{this.dbElement}}</button>
```
[the remark this answers is not in the document]
Comment (lanzoninicola) [reply apps/studio/src/components/CoreTabs.vue:91]: Thank you for all your comments, I will work on it in the next few days....

**Comment by lanzoninicola** [comment]
> 

I'm glad you like it. If all goes well, I should be able to let you know next week when the updated version is ready for your review.

**Comment by lanzoninicola** [comment]
Hello @rathboma,
I have finished all the tasks that were assigned to me and have also included the tests as per your suggestion. If you are ready to move forward with the review, I would greatly appreciate it.
Thank you.

**Review by rathboma — APPROVED** [review]
Amazing! Sorry it took so long, it's been a rough Q1 :-)

**Pull request merged**